simcon_project/templates/create_assignment.html

```diff
@@ -0,0 +1,286 @@
+                      window.alert("Adding failed, the template is already in the table.");
+                  }
+                  else {
+                      add2table(tbl1, name, tempTmp.creation_date, 2);
+                      addedTmp.push(key);
+                  }
+              }
+          }
+          if(find==false){
+              window.alert("Cannot find: "+name);
+          }
+      }
+
+      function add_label(tbl, name){
+          var find=false;
+          for(var i=0;i<subLabels.length;i++){
+              var labelTmp=subLabels[i].fields;
+              if(labelTmp.label_name==name){
+                  find=true;
+                  if(addedLabel.indexOf(name)>-1){
+                      window.alert("Adding failed, the label is already in the table.");
+                  }
+                  else {
+                      var researcher = labelTmp.researcher;
+                      add2table(tbl2, name, researcher, 3);
+                      addedLabel.push(name);
+                  }
+              }
+          }
+          if(find==false){
+              window.alert("Cannot find: "+name);
+          }
+      }
+
+      function verify_researcher(email){
+          for(var i=0;i<researchers.length;i++){
+              var researcherTmp=researchers[i];
+              if(researcherTmp.email==email){
+                  return true;
+              }
+          }
+          return false;
+      }
+
+      //Pass the client data to django through ajax.
+      function sub(){
+          $.ajaxSetup({
+              data: {csrfmiddlewaretoken: '{{ csrf_token }}' },
+          });
+          assign_name=document.getElementById("assign_name").value;
+          assign_date=document.getElementById("assign_date").value;
+          researcher_email=document.getElementById("researcher_email").value;
+          stuJSON=JSON.stringify(addedStu);
+          tempJSON=JSON.stringify(addedTmp);
+          labelJSON=JSON.stringify(addedLabel);
+
+          if(!verify_researcher(researcher_email)){
+              window.alert(researcher_email+" is not a researcher!");
+              return;
+          }
+
+          $.ajax({
+              url:"{% url 'assignments:add_data' %}",
+              type:'POST',
+              data: {
+                  'name': assign_name,
+                  'date': assign_date,
+                  'researcher': researcher_email,
+                  'stuData': stuJSON,
+                  'tempData': tempJSON,
+                  'labelData': labelJSON
+              },
+              dataType:'json',
+              success: function(data){
+                  if(data.success==0){
+                      window.alert("Submitted successfully!");
+                      location.reload();
+                  }
+                  else{
+                      var msgTmp=data.msg;
+                      window.alert(msgTmp);
+                  }
+              }
+          })
+      }
+  </script>
+</html>
```

Comment:
would love if you could comment the functions since its a lot of functions or if you could please have a js file!!